File: test-services/typescript/template/src/services/check.ts
import * as service from '../spec/check'
import * as models from '../spec/models'
import * as errors from '../spec/errors'

export let checkService = (): service.CheckService => {
    let checkEmpty = async (): Promise<void> => {}

    let checkEmptyResponse = async (params: service.CheckEmptyResponseParams): Promise<void> => {}
    
    let checkForbidden = async (): Promise<service.CheckForbiddenResponse> => {
        return {status: 'forbidden'}
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }

    let checkBadRequest = async (): Promise<service.CheckBadRequestResponse> => {
        return {status: 'bad_request', data: {message: 'Error returned from service implementation', location: errors.ErrorLocation.UNKNOWN}}
    }

    return {checkEmpty, checkEmptyResponse, checkForbidden, sameOperationName, checkBadRequest}
}

File: test-services/go/template/services/check.go
package services

import (
	"the-service/spec/check"
	"the-service/spec/empty"
	"the-service/spec/httperrors/errmodels"
	"the-service/spec/models"
)

type CheckService struct{}

func (service *CheckService) CheckEmpty() error {
	return nil
}

func (service *CheckService) CheckEmptyResponse(body *models.Message) error {
	return nil
}

func (service *CheckService) CheckForbidden() (*check.CheckForbiddenResponse, error) {
	return &check.CheckForbiddenResponse{Forbidden: &empty.Value}, nil
}

func (service *CheckService) SameOperationName() (*check.SameOperationNameResponse, error) {
	return &check.SameOperationNameResponse{Ok: &empty.Value}, nil
}

func (service *CheckService) CheckBadRequest() (*check.CheckBadRequestResponse, error) {
	badRequest := errmodels.BadRequestError{"Error returned from service implementation", errmodels.ErrorLocationUnknown, nil}
	return &check.CheckBadRequestResponse{BadRequest: &badRequest}, nil
}
